test/tests/api/v2_0/skupack_tests.py

```diff
@@ -0,0 +1,420 @@
+from on_http_api2_0.rest import ApiException
+from proboscis.asserts import *
+from proboscis import SkipTest
+from proboscis import test
+from json import dumps, loads
+from on_http_api2_0 import rest
+import time
+import requests
+import os
+import subprocess
+import json
+import tarfile
+import shutil
+
+
+LOG = Log(__name__)
+
+
+@test(groups=['skus_api2.tests'])
+class SkusTests(object):
+    def __init__(self):
+
+        self.__client = config.api_client
+        self.clear()
+        self.__nodes= []
+        self.__sku_id = ""
+        self.__workflows = {}
+        self.__tasks ={}
+        self.__config_json = {}
+        self.__rootDir ="/tmp/tarball/"
+        self.__skuPackTarball =  self.__rootDir +"mytest.tar.gz"
+
+    def __get_data(self):
+        return loads(self.__client.last_response.data)
+
+    @test(groups=['api2_get-skus'])
+    def test_skus(self):
+        """Testing GET:api/2.0/skus to get list of skus"""
+        Api().skus_get()
+        rsp = self.__client.last_response
+        assert_not_equal(404, rsp.status, message=rsp.reason)
+        assert_equal(200, rsp.status, message=rsp.reason)
+        data = loads(self.__client.last_response.data)
+
+
+    @test(groups=[ 'api2_post_skus'], depends_on_groups=['api2_check-skus'])
+    def post_sku(self):
+        """Test POST:api/2.0/skus"""
+        sku =   {
+          "name": "Quanta-D44",
+          "rules": [
+            {
+              "path": "dmi.Base Board Information.Manufacturer",
+              "contains": "Intel"
+            },
+            {
+              "path": "ohai.dmi.memory.total",
+              "equals": "32946864kB"
+            }
+          ],
+          "discoveryGraphName": "Graph.InstallCoreOS",
+          "discoveryGraphOptions": {
+            "username": "testuser",
+            "password": "hello",
+            "hostname": "mycoreos"
+          }
+        }
+        Api().skus_post(sku)
+        result = self.__client.last_response
+        data = loads(self.__client.last_response.data)
+        assert_equal(201, result.status, message=result.reason)
+        for key in sku.keys():
+                assert_equal(sku[key], data[key])
+
+        #POST the same SKU again and make sure that we get a 409
+        try:
+            Api().skus_post(sku)
+        except ApiException as e:
+            assert_true(409, e.status)
+
+    @test(groups=['api2_get_skus_by_id' ], depends_on_groups=['api2_post_skus'])
+    def test_skus_id(self):
+        """Testing GET:api/2.0/skus/id to get specific catalog details"""
+        Api().skus_get()
+        skus = loads(self.__client.last_response.data)
+        codes = []
+        for n in skus:
+            self.__sku_id  = n.get('id')
+            assert_is_not_none(self.__sku_id)
+            Api().skus_id_get(identifier=self.__sku_id)
+            rsp = self.__client.last_response
+            assert_not_equal(404, rsp.status, message=rsp.reason)
+            codes.append(rsp)
```

Comment:
Rather than appending the code and then retesting below, you could just test for the 200 here rather than testing for not_equal 404 (which cannot happen because an exception would have been thrown)